### PR TITLE
Bootstrap script for local development

### PR DESCRIPTION
**Pull request by hktouw**
resolves https://github.com/eastbaydsa/website/issues/241

## Why

The onboarding docs are a little outdated and the setup process is fairly manual. To streamline getting new developers on to the project, this PR adds a few scripts to simplify and speed up the onboarding process.

## How

This PR creates the following scripts:
  - `script/bootstrap`
    - Installs postgresql and rbenv
    - Installs project's current ruby version
    - Installs bundler
    - Installs gem dependencies
  - `script/setup`
    - Setups database and restores from postregsql dump
  - `script/server`
    - Boots up application

Additionally this PR / scripts handle several issues I ran into when trying to install the application.
  - Script handles issue where bundler fails to install gems because the project requires the specific bundler version the project was bundled with to be installed and not the default / current version.
  - Script shows error messaging when postgresql fails to start because data files from an older version of postgresql are present and need to be migrated.
  - This PR fixes a bug where database setup was failing because a database seed was referencing a model that has long since been deleted.

## Considerations

I have only tested this script on two computers, so there are undoubtedly things that may not work or can be improved. As new people onboard and use the script I can open up follow up PRs for any bugs that are reported.

This script only works on OSX since I am not familiar with how to setup a rails app in windows or linux.

### CODE REVIEW

**Comment by hktouw** [comment]
I opened this PR, but then saw Max's comment here https://github.com/eastbaydsa/website/issues/241#issuecomment-578456721 suggesting we include importing the postgresql data dump in the setup script. 

Consider this PR still WIP, but ready for review after I add that step.

File: README.md [hunk 1]
```diff
@@ -7,44 +7,28 @@ going forward.
...
 ## Getting Started
 
-Dependencies:
-
-- Ruby >= 2.4.0
-- Postgres
-  - [postgresapp.com](https://postgresapp.com) is a great option here
-  - Alternatively, you can install with [homebrew](https://brew.sh/)
-- bundler (gem install bundler)
-- foreman (gem install foreman)
```
Comment (BrainScraps) [inline README.md]: I like the concept of having this all automated, but also if someone who wants to contribute is joining us on a Windows or Linux machine, I think this section would still be useful to them. So maybe it's worth saying something like "the bootstrap script will give you the following dependencies if you're on Mac OS"

Comment (hktouw) [reply README.md]: Went ahead and reverted my deletions here https://github.com/eastbaydsa/website/pull/242/commits/13bcc674ee595f39c7eae9a8bffd876268d33d0a

This PR should only add a new section called `Quick start for OSX`

**Comment by hktouw** [comment]
What is the process for deploying / merging PRs? Are these instructions up to date? https://github.com/eastbaydsa/website#deploy

**Comment by maxlapides** [comment]
> What is the process for deploying / merging PRs? Are these instructions up to date? https://github.com/eastbaydsa/website#deploy

I'm not sure if these instructions still work. Normally the process it any changes to master are automatically pushed to our staging site, and then we manually deploy to production. I just added you to our Heroku staging instance so you can check it out.

**Comment by hktouw** [comment]
Sounds good, so to confirm that mean I should just merge this to master and we should be good?

**Comment by maxlapides** [comment]
@hktouw I think we're good to merge to master! The Heroku deployment shouldn't be affected.

**Comment by tilgovi** [comment]
This is really nice!

**Pull request merged**